helios-services/src/main/java/com/spotify/helios/master/ZooKeeperMasterModel.java

```diff
@@ -459,14 +480,21 @@ public void updateDeploymentGroupHosts(final String groupName, final List<String
         final List<ZooKeeperOperation> ops = Lists.newArrayList();
         ops.add(set(Paths.statusDeploymentGroupHosts(groupName), Json.asBytes(hosts)));
 
-        final DeploymentGroup deploymentGroup = getDeploymentGroup(groupName);
+        DeploymentGroup deploymentGroup = getDeploymentGroup(groupName);
         ImmutableList<Map<String, Object>> events = ImmutableList.of();
 
         if (deploymentGroup.getJobId() != null) {
-          final DeploymentGroupStatus deploymentGroupStatus = getDeploymentGroupStatus(groupName);
-          if (deploymentGroupStatus == null || deploymentGroupStatus.getState() != FAILED) {
-            final RollingUpdateOp op =
-                getInitRollingUpdateOps(deploymentGroup, hosts, HOSTS_CHANGED, client);
+          if (updateOnHostChange(deploymentGroup)) {
```

Comment:
Maybe only trigger roling-updates if the current state is `FAILED` or `SUCCEEDED` (not `ROLLING_OUT`)?

If I understand correctly, you can now time things to get perhaps unexpected results:
 - trigger a manual rolling-update of a bad, broken job
 - immediately after, before the rolling-update fails, a new hosts is added
 - a roll out (with reason=HOSTS_CHANGED) is triggered and fails
 - more hosts are added...
 - this triggers a rolling-update because the last reason had reason=HOSTS_CHANGED